Curry's paradox leads to clang error in c backend

Taken from: https://counterexamples.org/currys-paradox.html

```
ex is
  Curry ref is
    r(x Curry) i32 is abstract

  my_loop i32 is
    c := (ref : Curry
      r(x Curry) => x.r x
    )
    c.r c

  _ := my_loop
```
```
error 1: java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.base/java.util.Objects.checkIndex(Objects.java:361)
	at java.base/java.util.ArrayList.get(ArrayList.java:427)
	at dev.flang.be.c.C.tailRecursion(C.java:1198)
	at dev.flang.be.c.C.call(C.java:1119)
	at dev.flang.be.c.C.access(C.java:901)
	at dev.flang.be.c.C$CodeGen.call(C.java:177)
	at dev.flang.be.c.C$CodeGen.call(C.java:62)
...
```

```
/tmp/fuzion_ex_12076878201904451167.c:101:75: error: parameter 'fzouter' set but not used [-Werror,-Wunused-but-set-parameter]
fzT_1i32 fzC__L36ex__my_u_l__nymous0__1r(fzT_ex__my_u_loop_R_Hanonymous0* fzouter, fzT_ex_RCurry* arg0)
                                                                          ^
1 error generated.

error 1: C backend: C compiler failed
C compiler call 'clang --target=x86_64-pc-linux-gnu -ftrapv -Wall -Werror -Wextra -Wpedantic -Wformat=2 -Wno-unused-parameter -Wshadow -Wwrite-strings -Wold-style-definition -Wredundant-decls -Wnested-externs -Wmissing-include-dirs -Wno-strict-prototypes -Wno-gnu-empty-initializer -Wno-zero-length-array -Wno-trigraphs -Wno-gnu-empty-struct -Wno-unused-variable -Wno-unused-label -Wno-unused-function -Wno-pointer-to-int-cast -Wno-missing-include-dirs -Wno-infinite-recursion -Wno-unused-but-set-variable -O3 -DGC_THREADS -DGC_PTHREADS -DPTW32_STATIC_LIB -DGC_WIN32_PTHREADS -fno-trigraphs -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -std=c11 -o ex /home/not_synced/fuzion/build/include/shared.c /home/not_synced/fuzion/build/include/posix.c /tmp/fuzion_ex_12076878201904451167.c -lgc' failed with exit code '1'

one error.

```

--- a/src/dev/flang/be/c/C.java
+++ b/src/dev/flang/be/c/C.java
@@ -754,6 +754,7 @@ public class C extends ANY
           "-Wpedantic",
           "-Wformat=2",
           "-Wno-unused-parameter",
+          "-Wno-unused-but-set-parameter", // needed for #1777
           "-Wshadow",
           "-Wwrite-strings",
           "-Wold-style-definition",

--- a/tests/reg_issue1777/Makefile
+++ b/tests/reg_issue1777/Makefile
@@ -1,0 +1,25 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test Makefile
+#
+# -----------------------------------------------------------------------
+
+override NAME = reg_issue1777
+include ../compile.mk # infinite recursion

--- a/tests/reg_issue1777/reg_issue1777.fz
+++ b/tests/reg_issue1777/reg_issue1777.fz
@@ -1,0 +1,35 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test reg_issue1777
+#
+# -----------------------------------------------------------------------
+
+reg_issue1777 =>
+
+  Curry ref is
+    r(x Curry) i32 => abstract
+
+  my_loop i32 =>
+    c := (ref : Curry
+      redef r(x Curry) => x.r x
+    )
+    c.r c
+
+  _ := my_loop

